scipy/stats/_continuous_distns.py

```diff
@@ -3456,8 +3456,25 @@ def _munp(self, n, a, c):
         return sc.poch(a, n*1.0/c)
 
     def _entropy(self, a, c):
-        val = sc.psi(a)
-        return a*(1-val) + 1.0/c*val + sc.gammaln(a) - np.log(abs(c))
+        def regular(a, c):
+            val = sc.psi(a)
+            A = a * (1 - val) + val / c
+            B = sc.gammaln(a) - np.log(abs(c))
+            h = A + B
+            return h
+
+        def asymptotic(a, c):
+            # gammaln(a) ~ a * ln(a) - a - 0.5 * ln(a) +
+            #              0.5 * ln(2 * pi) + 1 / (12 * a)
+            # psi(a) ~ ln(a) - 1 / (2 * a) - 1 / (12 * a ^ 2)
+            P = np.log(a) - 1 / (2 * a) - 1 / (12 * a ** 2)
+            A = 0.5 + 1 / (6 * a) + 1 / (c * P) - 0.5 * np.log(a)
+            B = 0.5 * np.log(2 * np.pi) - np.log(abs(c))
+            h = A + B
```

Comment:
I cannot really follow here. Especially the  `1/(c*P)` terms looks like an asymptotic expansion of the misinterpreted `1.0/(c*val)` from before. Could you recheck?

Reply:
>Especially the 1/(c*P) terms looks like an asymptotic expansion of the misinterpreted 1.0/(c*val) from before

Yes, you're right. This slipped past the tests because my mpmath implementation was based on the same misunderstanding as my asymptotic expansion. I've now corrected it and simplified the expansion with the normal entropy. As for how the formula was derived, here's a writeup:
[Gengamma_entropy.pdf](https://github.com/scipy/scipy/files/10962107/Gengamma_entropy.pdf)